### DISCUSSION TITLE
Radio button vs. toggle switch (on multiple binaries page)

### DISCUSSION THREAD

**Discussion by cgolankotter** [twilio-labs/paste#2644] [Q&A] [answered]
Hi, We're updating a settings page for Authy that has **11** binary settings. (see "current" section in the screenshot)

Using Y/N radio buttons, the page gets pretty long with many floating yes/no radios.

Is it possible to use a single toggle switch instead?
This will allow us to show two more items and save scrolling with fewer "handles." (See "toggled" section in the screenshot)

What do you recommend?
Thanks!

<img width="2446" alt="current + toggled" src="https://user-images.githubusercontent.com/107283881/186945826-8234fce8-f5bc-4251-b4bc-963091c7e550.png">

**Comment by SiTaggart** [comment, marked as answer]
Hi @cgolankotter 

This type of use case seems totally reasonable for a switch!

The only thing I'd caution against is that the use of Switches is usually in conjunction with instant  on / off configuration and persistence. Think about how switches work on a phone in settings. There's no "save" action, they are just "on" or "off".

So if the API you're working with doesn't support asynchronous saving per field, and you have to rely on a save button for the entire form, then you might create an unfamiliar interaction for the user. In that instance, you would be better off using Radio's, even if they do create longer pages, because it's a much more familiar interaction pattern.

Hope that helps!

**Reply by cgolankotter** [reply]
Thanks so much Simon, this is super helpful!
